src/libcore/panic.rs

```diff
@@ -249,3 +249,13 @@ impl<'a> fmt::Display for Location<'a> {
         write!(formatter, "{}:{}:{}", self.file, self.line, self.col)
     }
 }
+
+/// An internal trait used by libstd to pass data from libstd to `panic_unwind`
+/// and other panic runtimes. Not intended to be stabilized any time soon, do
+/// not use.
+#[unstable(feature = "std_internals", issue = "0")]
+#[doc(hidden)]
+pub trait BoxMeUp {
```

Comment:
nit: shouldn't this be an `unsafe trait` since the `box_me_up` method must return a valid box pointer?